### PR TITLE
Add "telemetry" resources for logs

### PR DESCRIPTION
**Pull request by pellared**
## Why

I have found a bug when introducing MockSpansCollector and refactoring existing collectors.

Part of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/915

## What

- Add missing resources for logs
- Add tests for spans and logs resources
- Refactor the MockCollectors (introduce reusable "Expectors")

## Tests

CI

## Checklist

- [x] ~~`CHANGELOG.md` is updated.~~
- [x] ~~Documentation is updated.~~
- [x] New features are covered by tests.

### CODE REVIEW

File: test/IntegrationTests/Helpers/TestHelper.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public string GetTestAssemblyPath()
     /// and returns the Process instance for further interaction.
     /// </summary>
     /// <returns>Test application process</returns>
-    public Process StartTestApplication(int traceAgentPort = 0, int metricsAgentPort = 0, int logsAgentPort = 0, string arguments = null, string packageVersion = "", int aspNetCorePort = 0, string framework = "", bool enableStartupHook = true, bool enableClrProfiler = true)
+    public Process StartTestApplication(int traceAgentPort = 0, int otlpTraceCollectorPort = 0, int metricsAgentPort = 0, int logsAgentPort = 0, string arguments = null, string packageVersion = "", int aspNetCorePort = 0, string framework = "", bool enableStartupHook = true, bool enableClrProfiler = true)
```
Comment (Kielek) [inline test/IntegrationTests/Helpers/TestHelper.cs:69]: tracerAgentPort -> tracerZipkinAgentPort
otlpTraceCollectorPort -> tracerAgentPort 

What is more, I would move tracerZipkinAgentPort at postion 4 o even later.

File: test/IntegrationTests/Helpers/OtlpTracesSettings.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace IntegrationTests.Helpers;
+
+public class OtlpTracesSettings
```
Comment (Kielek) [inline test/IntegrationTests/Helpers/OtlpTracesSettings.cs:19]: TracesSettings -> Zipkin TracesSetting.
OtlppTracesSettings -> TracesSettings

All places where applicable.

**Review by Kielek — APPROVED** [review]

File: test/IntegrationTests/Helpers/TestHelper.cs (hunk 1 above)
Comment (pellared) [reply test/IntegrationTests/Helpers/TestHelper.cs:69]: I have not touched it as the end goal is to:

- otlpTraceCollectorPort  -> tracerAgentPort
- remove tracerZipkinAgentPort

File: test/IntegrationTests/Helpers/MockLogsCollector.cs [hunk 3]
```diff
@@ -165,42 +170,26 @@ public void AssertExpectations(TimeSpan? timeout = null)
...
+        var logsMessage = ExportLogsServiceRequest.Parser.ParseFrom(ctx.Request.InputStream);
+        foreach (var resourceLogs in logsMessage.ResourceLogs ?? Enumerable.Empty<ResourceLogs>())
         {
-            var logsMessage = ExportLogsServiceRequest.Parser.ParseFrom(ctx.Request.InputStream);
-            if (logsMessage.ResourceLogs != null)
+            ResourceExpector.Collect(resourceLogs.Resource);
+            foreach (var scopeLogs in resourceLogs.ScopeLogs ?? Enumerable.Empty<ScopeLogs>())
             {
-                foreach (var rLogs in logsMessage.ResourceLogs)
+                foreach (var logRecord in scopeLogs.LogRecords ?? Enumerable.Empty<LogRecord>())
```
[the remark this answers is not in the document]
Comment (pellared) [reply test/IntegrationTests/Helpers/MockLogsCollector.cs:179]: @Kielek Do you want me to change it in the following PR?

[the remark this answers is not in the document]
Comment (Kielek) [reply test/IntegrationTests/Helpers/MockLogsCollector.cs:179]: It is up to you. I wouldn't write the code in this way, but I can live with it :)

**Pull request merged**